vendor/blockchain_contracts/src/ethereum/mod.rs

```diff
@@ -1,13 +1,58 @@
-use ethereum_support::Bytes;
-use serde::{Deserialize, Serialize};
+use crate::rfc003::{secret_hash::SecretHash, timestamp::Timestamp};
+use binary_macros::{base16, base16_impl};
+use byteorder::{BigEndian, WriteBytesExt};
+use web3::types::{Address, Bytes, U256};
 
 pub mod rfc003;
 
-#[derive(Deserialize, Serialize, Debug)]
-pub struct ByteCode(pub String);
+#[derive(Debug)]
+pub enum ToEvmError {
+    Io(std::io::Error),
+}
+
+pub trait EncodeToEvm {
+    fn encode_to_evm(&self) -> Result<Vec<u8>, ToEvmError>;
+}
+
+impl EncodeToEvm for SecretHash {
+    fn encode_to_evm(&self) -> Result<Vec<u8>, ToEvmError> {
+        Ok(self.clone().into())
+    }
+}
+
+impl EncodeToEvm for Address {
+    fn encode_to_evm(&self) -> Result<Vec<u8>, ToEvmError> {
+        Ok(self.to_vec())
+    }
+}
+
+impl EncodeToEvm for U256 {
+    fn encode_to_evm(&self) -> Result<Vec<u8>, ToEvmError> {
+        let mut vec = vec![0; 32];
+        self.to_big_endian(&mut vec);
+        Ok(vec)
+    }
+}
 
-impl Into<Bytes> for ByteCode {
-    fn into(self) -> Bytes {
-        Bytes(hex::decode(self.0).unwrap())
+impl EncodeToEvm for Timestamp {
+    fn encode_to_evm(&self) -> Result<Vec<u8>, ToEvmError> {
+        let mut vec = vec![];
+        vec.write_u32::<BigEndian>(self.clone().into())
+            .map_err(ToEvmError::Io)?;
```

Comment:
If you  don't want to do an `expect` here, you can get rid of the error by doing:

```
let mut buf = [0; 4];
BigEndian::write_u32(&mut buf, 1_000_000);
buf.to_vec()
```
This looks better anyway.

:x: This stuff shouldn't return a `Result`.